meson.build

```diff
@@ -112,11 +112,107 @@ if get_option('manta')
   add_project_arguments('-DSAPF_MANTA', language: 'cpp')
 endif
 
+release_cpp_args = cpp_args + ['-O3']
```

Comment:
if im reading this correctly, it seems we are now always building with optimizations. but it would be good to have this depend on the build type (`debug`/`release`, perhaps others like `debugoptimized`). i have not yet looked into how Meson deals with this, which is why i left optimizations out initially 😄

Reply:
Ah okay, I think we are supposed to be using this `buildtype` param for this anyway, at least partially. I also didn't know how they worked. I think they automatically set some of the needed params. I will rework things to try to use them properly...

Apparently here's how they work, for reference:
- debug is the default buildtype. It adds debug symbols (`-g`) and disables optimizations (`-O0`)
- release adds `-O3` and `-DNDEBUG` to disable asserts, and omits debug symbols.
- debugoptimized adds `-O2` but keeps debug info.
- plain means we set our own flags and doesn't do anything else

Reply:
Note that this means the convention for meson is that `debug` is the default buildtype. So if we want to stick to that convention, I think we'll need to make it so that the unoptimized build is the default, and the user would need to manually set the buildtype to build for a release. 

IMO it's probably best to stick to that convention to minimize confusion for those familiar with meson. But if we really want to, we could make `release` the default buildtype via meson.options. WDYT?

Reply:
debug-as-default is a common convention, and i prefer it personally, so no objections :)